### ISSUE TITLE
AQL: cannot and-combine normalized condition

### ISSUE THREAD

**Issue by aquamatthias** [arangodb/arangodb#19175]
I updated my ArangoDB installation to 3.11 and now I see queries failing with this message:

```
Error: AQLQueryExecuteError
Message: [HTTP 500][ERR 4] AQL: cannot and-combine normalized condition (exception location: /Users/jenkins/hw22-mac-m1/oskar/work/ArangoDB/arangod/Aql/Condition.cpp:668). Please report this error to arangodb.com (while optimizing plan) (exception location: /Users/jenkins/hw22-mac-m1/oskar/work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:279). Please report this error to arangodb.com
```

The same queries worked in older versions of ArangoDB so I consider this a regression.

## My Environment

* __ArangoDB Version__:        3.11.0
* __Deployment Mode__:      Single Server
* __Deployment Strategy__:  Manual Start
* __Operating System__:        MacOS 13.4
* __Total RAM in your machine__:        16GB
* __Disks in use__:  SSD

## Component, Query & Data

__Affected feature__:
AQL

__AQL query (if applicable)__:
```
LET filter0 = (FOR m0 in `resoto` FILTER @b0 IN m0.kinds  RETURN m0)LET m1 =( FOR c0_0 in filter0 FOR c0_1 IN 0..1 OUTBOUND c0_0 resoto_default OPTIONS { bfs: true, uniqueVertices: 'global' } FILTER (c0_0._key==c0_1._key) or ((@b1 IN c0_1.kinds) and (c0_1.trail_status.is_logging == @b2)) COLLECT l0_l0_res=c0_0 WITH COUNT INTO counter1 FILTER counter1==0  RETURN l0_l0_res)  FOR result in m1 RETURN UNSET(result, ["flat"])
```
with following bind vars:
```json
{
        "b0": "aws_region",
        "b1": "aws_cloud_trail",
        "b2": true
}
```

__AQL explain and/or profile (if applicable)__:

```
Error: AQLQueryExecuteError
Message: [HTTP 500][ERR 4] AQL: cannot and-combine normalized condition (exception location: /Users/jenkins/hw22-mac-m1/oskar/work/ArangoDB/arangod/Aql/Condition.cpp:668). Please report this error to arangodb.com (while optimizing plan) (exception location: /Users/jenkins/hw22-mac-m1/oskar/work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:279). Please report this error to arangodb.com
```

__Dataset__:

It is a simple graph: one vertex collection with multiple edge collection.
I don't think the data is the issue, but I can share the data if required.

__Size of your Dataset on disk__:

tiny < 1GB.

__Replication Factor & Number of Shards (Cluster only)__:

none

## Steps to reproduce

1. Type above query and bind vars into AQL console.

__Problem__:

Regression in 3.11 - the query is not executed. 
The same query worked in 3.8, 3.9 and 3.10.

__Expected result__:

The query can be executed.

**Comment by hkernbach** [comment]
Thank you @aquamatthias for your contribution. We are able to reproduce the issue. We're on it! As you already expected, the issue is not related to your data.

**Comment by jsteemann** [comment]
Fixed via https://github.com/arangodb/arangodb/pull/19264, targeting 3.11.2

**Comment by asifkazi** [comment]
This is fixed in 3.11.2 which was released yesterday
